### PR TITLE
Fix collapsed threads not restoring, and the clock skew correction

### PR DESCRIPTION
**Pull request by paskal**
Three defects in the widget, found while building the Go e2e suite.

## Collapsed threads did not come back

Collapse state was kept as a flat list of `siteID_url_commentID` strings and read back by splitting on `_`. Any underscore in the url, the site id or the comment id made the pieces impossible to tell apart, so a page whose url contains one lost its collapsed threads on every reload.

The same ambiguity let one page's entries be read or deleted as another's: `/post` matched everything stored for `/post_2`, returning a phantom comment id and, on the next save, deleting the other page's state outright. A site id of `blog` collided with `blog_ru` by the same rule, and every site embedding one instance shares that storage.

No separator fixes this, since any character can occur inside the values. The ids are now nested under the site and then the url, so nothing has to be taken apart again. Anything stored in the old shape reads as empty: collapsed threads are a view preference, and re-expanding them once is not worth a migration.

The e2e suite had been stripping underscores out of its own thread urls to work around this, which left its collapse test unable to fail on the bug it covers. That workaround is gone, and the test now fails without this fix.

## The clock skew correction was a thousandth of the skew

`serverClientTimeDiff` was written in seconds and added to an epoch in milliseconds, so the correction it exists to apply barely moved the edit deadline. It is now milliseconds, and named for the unit.

A response with no usable `date` header used to fall back to a zero timestamp, which already made the "skew" about twenty days and would have made it fifty-five years once the units were right. Nothing is stored now unless the reading is plausible, since `Date.parse` is lenient enough to turn junk into a date and let an absurd value through the branch that parses rather than the one that rejects.

## The controversy tooltip was dead

The score tooltip reports controversy again when there is any. It has been dead since the vote component was rewritten in 0e4ae6e0, which moved the score into its own component and left the line behind commented out; the value has been passed in and dropped ever since, while the string stayed in all 24 catalogues.

Unlike the original it stays out of the way when there is no controversy, which the backend sends as an absent field rather than a zero.

## Also here

`frontend/.dockerignore` is deleted. Docker reads `.dockerignore` from the build context root only, and nothing builds from `frontend/`: every context in the repo is the repository root, apart from the site, which has its own. Both lines it carried are already in the root `.dockerignore` verbatim.

## Tests

Every fix has tests that fail without it, checked by reverting each one individually rather than assumed. The collapse tests cover the sibling-url and site-id collisions and the old-format read; the skew tests cover the unit, the missing header and the implausible reading; the tooltip tests cover present, zero and absent controversy.

### CODE REVIEW

File: frontend/apps/remark42/app/store/thread/utils.ts [hunk 1]
```diff
@@ -3,30 +3,40 @@ import { LS_COLLAPSE_KEY } from 'common/constants';
...
  */
 export const saveCollapsedComments = (siteId: string, url: string, info: Comment['id'][]): void => {
-  const data = info.map((i) => `${siteId}_${url}_${i}`);
-  const notForThisPost = getFromLocalStorage().filter((entry) => entry.indexOf(`${siteId}_${url}`) === -1);
-  const all = new Set([...notForThisPost, ...data]);
-  localStorageSetItem(LS_COLLAPSE_KEY, JSON.stringify([...all]));
+  const stored = getFromLocalStorage();
+
+  localStorageSetItem(LS_COLLAPSE_KEY, JSON.stringify({ ...stored, [siteId]: { ...stored[siteId], [url]: info } }));
 };
```
[the remark this answers is not in the document]
Comment (paskal) [reply frontend/apps/remark42/app/store/thread/utils.ts:59]: Done. An empty list now drops the url entry, and the site entry with it when that was the last page under it, so collapsing and re-expanding leaves nothing behind. Two cases in `utils.test.ts` cover it.

File: frontend/apps/remark42/app/store/thread/utils.ts [hunk 2]
```diff
@@ -3,30 +3,40 @@ import { LS_COLLAPSE_KEY } from 'common/constants';
...
  */
-export const getCollapsedComments = (): string[] =>
-  getFromLocalStorage().reduce((acc: string[], v: string) => {
-    const components = v.split('_');
-    if (components[0] === siteId && components[1] === url) {
-      acc.push(components[2]);
-    }
-    return acc;
-  }, []);
+export const getCollapsedComments = (): Comment['id'][] => getFromLocalStorage()[siteId]?.[url] ?? [];
```
[the remark this answers is not in the document]
Comment (paskal) [reply frontend/apps/remark42/app/store/thread/utils.ts]: Done. `getCollapsedComments` now returns the stored value only when it is an array and an empty list otherwise, which is what the reducer needs. The top level was already checked, so the leaf being unchecked was an inconsistency. Covered by a test storing a string in place of the id list.

File: frontend/apps/remark42/app/common/static-store.ts [hunk 3]
```diff
@@ -3,7 +3,8 @@ import type { Config } from './types';
 interface StaticStoreType {
   config: Config;
   /** used in fetcher, fer example to set comment edit timeout */
```
[the remark this answers is not in the document]
Comment (paskal) [reply frontend/apps/remark42/app/common/static-store.ts]: Fixed, along with the line the typo was on: two doc comments had been left stacked on the same field, and the stale one described the units this change moved away from. There is one comment there now.

**Review by umputun — APPROVED** [review]
approving. All three hold up, and the collapse one is a worse bug than the summary makes it sound: losing state on a url with an underscore is the mild half, and one page deleting another's is the half that would have been hard to explain from a report.

the new read path is total for every shape that parses. The flat array is rejected at the top level, a missing site, `{site: null}`, a primitive at the site key and a site with no entry for the current url all reach `undefined` through `stored[siteId]?.[url]`, and a leaf of the wrong type is caught by `Array.isArray`. Nesting removes the ambiguity rather than relocating it, since site and url are separate exact lookups with no separator to parse and no prefix that can match a longer one.

the skew correction is right in every consumer. `new Date(props.data.time).getTime() + timeDiff + editDuration * 1000` is milliseconds throughout now, where `timeDiff` used to be seconds in that same sum. Renaming to `serverClientTimeDiffMs` is what makes that safe, since every reader had to be touched, and none was missed. A missing or unparseable `date` header stores nothing instead of falling back to epoch zero.

deleting `frontend/.dockerignore` is safe. Its two lines are in the root file verbatim, and nothing uses `frontend/` as a build context.

**one line, and it predates this**

`utils.ts:17` parses without a guard, so malformed JSON in that key throws out of `restoreCollapsedThreads` on page load. Master has the same parse, so this PR does not open it. `getJsonItem` in `common/local-storage.ts` already wraps exactly this in a try and returns null, and this change is otherwise about making that function total against whatever is in storage, so it is a natural thing to pick up while in there.
